shared/reducers/crypto.tsx

```diff
@@ -102,44 +113,75 @@ export default Container.makeReducer<Actions, Types.State>(initialState, {
     } = action.payload
     if (operationGuard(operation, action)) return
 
-    // Bail if the user has cleared the input before the RPC has returned a result
-    if (!draftState[operation].input.stringValue()) {
-      return
+    let inputAction:
+      | CryptoGen.SaltpackDecryptPayload
+      | CryptoGen.SaltpackEncryptPayload
+      | CryptoGen.SaltpackSignPayload
+      | CryptoGen.SaltpackVerifyPayload
+      | undefined
+
+    switch (input?.type) {
+      case CryptoGen.saltpackDecrypt:
+      case CryptoGen.saltpackEncrypt:
+      case CryptoGen.saltpackSign:
+      case CryptoGen.saltpackVerify:
+        inputAction = input
+        break
+      default:
+        inputAction = undefined
+    }
+
+    let outputMatchesInput = false
+
+    const op = draftState[operation]
```

Comment:
the new logic. We keep track of the output matching the input. if the old output matches the input (we already have the correct response, then we ignore an out of order response), otherwise we update the output , but keep track that its still not good (yet)